src/main/scala/strawman/collection/immutable/Iterable.scala

```diff
@@ -0,0 +1,40 @@
+package strawman.collection.immutable
+
+import scala.{Any, Boolean, Int}
+import scala.util.hashing.MurmurHash3
+
+/**
+  * Base class of immutable collections. (Must *not* be inherited by mutable collections)
+  */
+trait Iterable[+A]
+  extends strawman.collection.Iterable[A]
+  with strawman.collection.IterableLike[A, Iterable] {
```

Comment:
Note that the `immutable.Iterable` trait inherits from the “mutability-agnostic” trait `Iterable`. I don’t think this is a problem since this trait only provides read operations.